provider.go

```diff
@@ -143,26 +122,20 @@ func (p *Provider) DeleteRecords(_ context.Context, zone string, records []libdn
 
 	for _, record := range records {
 		var queuedDeletes []libdns.Record
-		if record.ID == "" {
-			// Try fetch record in case we are just missing the ID
-			matches, err := p.getMatchingRecord(record, zone)
-			if err != nil {
-				return deletedRecords, err
-			}
-			for _, rec := range matches {
-				queuedDeletes = append(queuedDeletes, rec)
-			}
-		} else {
-			queuedDeletes = append(queuedDeletes, record)
-		}
+		queuedDeletes = append(queuedDeletes, record)
 
 		reqJson, err := json.Marshal(credentials)
 		if err != nil {
 			return nil, err
 		}
 
 		for _, recordToDelete := range queuedDeletes {
-			_, err = MakeApiRequest(fmt.Sprintf("/dns/delete/%s/%s", trimmedZone, recordToDelete.ID), bytes.NewReader(reqJson), pkbnResponseStatus{})
+			rr := recordToDelete.RR()
+			trimmedName := rr.Name
+			if trimmedName == "@" {
+				trimmedName = ""
+			}
```

Comment:
Since we need some more robust conversion logic, I think this logic for handling the root record would be good to fold into that too, we're doing this in multiple places right now, which isn't ideal for maintenance.

Reply:
I'll refactor this out to a shared function

Reply:
Handled in [08eb76a](https://github.com/libdns/porkbun/pull/14/commits/08eb76a9d8b3a6033325ff44165622eabd653973)